### PR TITLE
SCMOD-8597: Updated log messages

### PR DESCRIPTION
**Pull request by Anthony-McGreevy**
**Jira**: https://portal.digitalsafe.net/browse/SCMOD-8597
**Build**: http://sou-jenkins2.swinfra.net/job/Autoscaler/view/Developer/job/Autoscaler~autoscaler~SCMOD-8597~CI/
Example log output attached to Jira ticket, log collected from larry system under load.

### CODE REVIEW

**Review by michael-bryson — APPROVED** [review]
This looks ok to me but lets have Rob take a look too, I can't add him to this PR though as he must not access to this repo.

File: autoscale-core/src/main/java/com/hpe/caf/autoscale/core/ScalerThread.java [hunk 1]
```diff
@@ -211,6 +214,7 @@ private void scaleDown(final int amount)
             throw new ScalerException("An error occured during an attempt to have the main thread sleep before rechecking the number"
                 + " of instance present for the application.", ex);
         }
+        LOG.debug("Service {} scaled down by {} instances.", serviceRef, amount);
```
Comment (rob-duewer) [inline autoscale-core/src/main/java/com/hpe/caf/autoscale/core/ScalerThread.java]: Why is this at debug level?

**Review by rob-duewer — CHANGES_REQUESTED** [review]
I apparently don't have write access to this public repo, but yes, I can see the content.

It generally looks OK to me.  But I am concerned that you're eliminating the log entries at the info level; particularly on the scale down event.  We need those to be recorded at the standard prod level so we know not only when AS is scaling up, but also when it's actively scaling down.  (This allows us to determine when an instance fell off because it crashed or was otherwise terminated, and one stopped by AS normally.)  Scaling down events that would have been triggered when the total is already at the minimum do not need to be recorded.

I'm also concerned that at prod log level, this is not actually recording the case described in the ticket. What we need to know in the short log entry, that the scaling up event would have added X instances, but because of the configured maximum(Y); that was not possible; only Z instances were added to the count.  And Z can be zero, if the app is at its configured maximum.

The reason we need this (and at the info level) is so that we can parse the log if desired and look for these entries.  From an ops perspective, what this tells us is whether or not we have allocated sufficient resources to the cluster and set the maximum values for the app properly.  If we see a log line like this, we care very much about its frequency and the relative values of X, Y, and Z.  As an operator, this tells us how much pressure we have, allowing us to consider resource addition or other reconfiguration.

File: autoscale-core/src/main/java/com/hpe/caf/autoscale/core/GovernorImpl.java [hunk 2]
```diff
@@ -151,23 +168,39 @@ public ScalingAction govern(String serviceRef, ScalingAction action, final int c
...
+                            + "minimum number instances, due to the scaling restriction this will be reduced to attempt to allow other "
+                            + "services to reach their minimum. The scaling command has been overriden to scale the service "
+                            + "down. New command is SCALE_DOWN by {} instances", serviceRef, amount);
                         return new ScalingAction(ScalingOperation.SCALE_DOWN, amount);
                     }
                 } else {
                     final int delta = Math.min(scalingConfiguration.getMaxInstances() - lastInstanceInfo.getTotalRunningAndStageInstances(),
                                                 Math.max(0, action.getAmount()));
+                    if(delta < action.getAmount()){
+                        LOG.info("Service {} requested {} more instances but will only be scaled by {} as the service has a max instance "
```
Comment (Anthony-McGreevy) [inline autoscale-core/src/main/java/com/hpe/caf/autoscale/core/GovernorImpl.java:182]: @rob-duewer I have added the message you have requested at info level.

**Pull request merged**